lib/sycamore/sycamore/connectors/opensearch/opensearch_reader.py

```diff
@@ -93,7 +98,17 @@ class OpenSearchReaderQueryResponse(BaseDBReader.QueryResponse):
     def to_docs(self, query_params: "BaseDBReader.QueryParams") -> list[Document]:
         assert isinstance(query_params, OpenSearchReaderQueryParams)
         result: list[Document] = []
-        if not query_params.reconstruct_document:
+        if query_params.doc_reconstructor is not None:
```

Comment:
How are you planning to implement parallel reconstruction? 
I think this is another place where making the reconstructor a class would help. It could have a member variable .parallel_reconstruct where the first stage does dict -> (simplified Doc); and the second stage is a Map to to simplified doc -> full doc.

Reply:
Maybe we can discuss this idea in #1029